dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DOIDataCiteRegisterService.java

```diff
@@ -96,45 +109,25 @@ public String registerIdentifier(String identifier, Map<String, String> metadata
             } else {
                 rc.setUrl(target);
             }
-            try {
-                DataCiteRESTfullClient client = getClient();
-                retString = client.postMetadata(xmlMetadata);
-                client.postUrl(identifier.substring(identifier.indexOf(":") + 1), target);
-            } catch (UnsupportedEncodingException ex) {
-                Logger.getLogger(DOIDataCiteRegisterService.class.getName()).log(Level.SEVERE, null, ex);
-            }
+            retString = registerDOI(identifier.split(":")[1], retString, target, xmlMetadata);
```

Comment:
This approach was already in other method 'identifier.substring(identifier.indexOf(":") + 1)' also if datacite id won't have 'doi:(...)' there will be exception in different place where id structure is checked.

Reply:
AFAIK `identifier.substring(identifier.indexOf(":")` will return -1 if the character is not found (of course it could still throw a NPE if identifier is null).
But if the exception will be thrown earlier then it's ok.

Reply:
My bad it is thrown when creating GlobalId in some cases, but in the code where the global id is created beforehand this method is utilized ```return protocol + ":" + authority + "/" + identifier;``` It is kind of a mess and I think Doi should be cleaned up but it is what it is for now.